src/mondeath.cpp

```diff
@@ -844,3 +844,19 @@ void mdeath::fireball( monster &z )
         normal( z );
     }
 }
+
+
+void mdeath::conflagration( monster &z )
+{
+    for( int x = z.pos().x - 1; x <= z.pos().x + 1; ++x ) {
+        for( int y = z.pos().y - 1; y <= z.pos().y + 1; ++y ) {
+            tripoint fire_tri = tripoint( x, y, z.pos().z );
+            g->m.propagate_field( fire_tri, fd_fire, 18, 3 );
+        }
+    }
+    std::string explode = string_format(
+                              _( "the %s belches sparks and flame erupts from it's stomach!" ),
```

Comment:
I'd rewrite the description: you might not see the eruption itself (only hear it), but the wording implies that you can actually see what's happening. Another option is adding a check like `g->u.sees( z.pos() )`.